Style engine: enqueue block supports styles in Gutenberg (#42880)
WordPress/gutenberg: packages/style-engine/README.md

The Style Engine powering global styles and block customizations.

## Backend API

### wp_style_engine_get_styles()

> Global public interface method to generate styles from a single style object, e.g., the value of

Reviewer: Tiny nit: interface and methods make me think of classes rather than a public function. Should we call it a public function instead?

```suggestion
Global public function to generate styles from a single style object, e.g., the value of
```

a [block's attributes.style object](https://developer.wordpress.org/block-editor/reference-guides/theme-json-reference/theme-json-living/#styles)
or
the [top level styles in theme.json](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/)
.

_Parameters_

-   _$block_styles_ `array` A block's `attributes.style` object or the top level styles in theme.json
-   _$options_ `array<string|boolean>` An array of options to determine the output.
    -   _context_ `string` An identifier describing the origin of the style object, e.g., 'block-supports' or '
        global-styles'. Default is 'block-supports'.

>     -   _enqueue_ `boolean` When `true` will attempt to store and enqueue for rendering on the frontend.

Reviewer: Possible clarification, feel free to ignore!

```suggestion
    -   _enqueue_ `boolean` When `true` will attempt to store and enqueue for rendering in a `style` tag on the site frontend.
```

    -   _convert_vars_to_classnames_ `boolean` Whether to skip converting CSS var:? values to var( --wp--preset--\* )
        values. Default is `false`.
    -   _selector_ `string` When a selector is passed, `generate()` will return a full CSS rule `$selector { ...rules }`,
        otherwise a concatenated string of properties and values.

_Returns_
`array<string|array>|null`

```php
array(
    'css'           => (string) A CSS ruleset or declarations block formatted to be placed in an HTML `style` attribute or tag.
    'declarations'  => (array) An array of property/value pairs representing parsed CSS declarations.
    'classnames'    => (string) Classnames separated by a space.
);
```

> It will return compiled CSS declartions for inline styles, or, where a selector is provided, a complete CSS rule.

Reviewer 2: ```suggestion
It will return compiled CSS declarations for inline styles, or, where a selector is provided, a complete CSS rule.
```

> To enqueue a style for rendering in the frontend, the `$options` array requires the following:

Reviewer: Possible clarification:

```suggestion
To enqueue a style for rendering in the site's frontend, the `$options` array requires the following:
```
